gcm_filters/kernels.py

```diff
@@ -12,12 +12,29 @@
 
 # not married to the term "Cartesian"
 GridType = enum.Enum(
-    "GridType", ["CARTESIAN", "CARTESIAN_WITH_LAND", "IRREGULAR_CARTESIAN_WITH_LAND"]
+    "GridType",
+    [
+        "CARTESIAN",
+        "CARTESIAN_WITH_LAND",
+        "IRREGULAR_CARTESIAN_WITH_LAND",
+        "POP_SIMPLE_TRIPOLAR_T_GRID",
```

Comment:
This kernel strikes me as very similar to the `CARTESIAN_WITH_LAND`, the only difference being the boundary condition. Do we want to think about harmonizing the names? What about `TRIPOLAR_CARTESIAN_WITH_LAND`?

Or maybe we want to rename things like `REGULAR`, `REGULAR_WITH_LAND`, `TRIPOLAR_REGULAR_WITH_LAND`, etc?

Reply:
Thanks, Ryan! Good point about harmonizing the names. I like your suggestion, and vote for: `REGULAR`, `REGULAR_WITH_LAND`, `IRREGULAR_WITH_LAND`, `TRIPOLAR_REGULAR_WITH_LAND`, `TRIPOLAR_IRREGULAR_WITH_LAND`. Or we could keep the latter as `POP_TRIPOLAR_T_GRID`, and change it later to something more generic (e.g., `TRIPOLAR_IRREGULAR_WITH_LAND`), as you suggested here: https://github.com/ocean-eddy-cpt/gcm-filters-paper/issues/18#issuecomment-790781694

Reply:
I think we should some distinguish between three types of grids:

- Regular + Rectangular: dx=dy=const, dimensions are globally orthogonal (what VTK calls "image data")
- Irregular + Rectangular, dimensions are globally orthogonal, but dx and dy can vary independently (what VTK calls "rectilinear"
- Curvilinear, dimensions are locally orthogonal, but the grid curves (what vtk calls "structure")
- Unstructured mesh (e.g. triangles, hexagons, etc.) - probably out of scope for now

In addition to this, we seem to have two independent option spaces
- Boundary conditions (e.g. singly or doubly periodic, tripolar, cubed-sphere, etc.)
- Land mask or not

(VTK reference: http://www.bu.edu/tech/support/research/training-consulting/online-tutorials/vtk/)

We don't need to resolve all of this now, but maybe we can use these concepts to inform the naming convention.

Down the road (future PR), we could maybe even use multiple inheritance to define kernels, like

```python
class POPCurvilinearGrid(CurvilinearGrid, GridWithLandMask, TripolarBoundaryCondition):
    pass
```

Reply:
Agreed, these are useful guidelines. 

The thing is that all Laplacians that we have implemented so far can be used on curvilinear grids, even the ones where dx=dy=1, e.g., `CARTESIAN_WITH_LAND` or `POP_SIMPLE_TRIPOLAR_T_GRID`. The latter are the Laplacians for fixed factor filtering, but they work for the POP **curvilinear** grid, for instance.

I like your idea about multiple inheritance to define kernels - this is more or less what I have imagined too. But I appreciate that this is maybe more of a long-term goal, and will come with its own challenges.

Reply:
Since all Laplacians are good for curvilinear grids, I decided to go with 
* "regular": Laplacian needs **no** grid info (for fixed factor filtering)
* "irregular": Laplacian needs grid info (for more general filtering, e.g., with fixed scale)

I'm happy to change this if there are better suggestions, either now or later.